### PR TITLE
chore: Automate deployment to try.vyperlang.org

### PR DESCRIPTION
**Pull request by DanielSchiavini**
### What I did
- Added a script that will connect to try.vyperlang.org and run the deployment script

### How I did it
- By using github actions

### How to verify it
- See [this successful action](https://github.com/DanielSchiavini/titanoboa/actions/runs/7913892279/job/21602463995) and how it nicely caches most of the work

### Description for the changelog
- Automated deployments to try.vyperlang.org

### Cute Animal Picture
image

### CODE REVIEW

File: .github/workflows/deploy.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+            User ${{ secrets.TRY_VYPER_USERNAME }}
+            IdentityFile ~/.ssh/try_vyper.key
+            StrictHostKeyChecking no
+          END
+
+      - name: Deploy
+        run: |
+          ssh try_vyper bash <<EOF
+          cd try.vyperlang.org
+          REF=${{ github.sha }} ./full_reload.sh
```
Comment (charles-cooper) [inline .github/workflows/deploy.yaml:31]: is the github sha needed? i think full_reload.sh should grab latest boa no?

Comment (DanielSchiavini) [reply .github/workflows/deploy.yaml:31]: By using the sha we can keep the docker cache

Comment (DanielSchiavini) [reply .github/workflows/deploy.yaml:31]: Basically it caches these lines: https://github.com/vyperlang/try.vyperlang.org/blob/c6148d8d66fe1a20a9883edb6de668cb2619f03b/notebook.Dockerfile#L4-L6

Comment (charles-cooper) [reply .github/workflows/deploy.yaml:31]: no more cache :smiling_face_with_tear:

**Pull request closed without merging**